### PR TITLE
In step-40, say at least a little bit about parallel linear algebra.

### PR DESCRIPTION
**Pull request by bangerth**
In contrast to many of the other early tutorial programs, step-40 has a rather spartan introduction. Add a little bit of context to it.

### CODE REVIEW

File: examples/step-40/doc/intro.dox [hunk 1]
```diff
@@ -93,6 +94,71 @@ It is probably worthwhile reading it for background information on how
 things work internally in this program.
 
 
+<h3>Linear algebra</h3>
+
+step-17 and step-18 already used parallel linear algebra classes, but since
+the current program is the first one that *really* covers parallel computing,
+it is probably worth giving a broad overview of paralel linear algebra here
```
Comment (kronbichler) [inline examples/step-40/doc/intro.dox:101]:
```suggestion
it is probably worth giving a broad overview of parallel linear algebra here
```

File: examples/step-40/doc/intro.dox [hunk 2]
```diff
@@ -93,6 +94,71 @@ It is probably worthwhile reading it for background information on how
...
+  the data among processes, or work on them on processes that cannot
+  access each other's memory directory. (On the other hand, many of
+  these classes actually use multiple threads internally, to use
+  the multiple processor cores available on today's laptops and
+  work stations.) These classes reside in the top-level
+  `namespace dealii`.
+- Interfaces to the PETSc library's implementations of linear
+  algebra functionality. These are found in namespace `PETScWrappers`.
+  PETSc is a library that has built a large collection of linear algebra,
+  linear solver, nonlinear solver, time stepper, and other functionality
```
Comment (kronbichler) [inline examples/step-40/doc/intro.dox:130]: Shouldn't these be plural forms here?
```suggestion
  linear solvers, nonlinear solvers, time steppers, and other functionality
```

File: examples/step-40/doc/intro.dox [hunk 3]
```diff
@@ -93,6 +94,71 @@ It is probably worthwhile reading it for background information on how
...
+  functionality Trilinos provides via its "Epetra" collection of vector
+  and matrix classes, and everything that builds on it. Epetra has been
+  slated for removal since the early 2020s (and may already have been removed
+  by the time you read this), with replacements provided in the "Tpetra"
+  collection. Tpetra replaces the old classes by ones that are templatized
+  on the type of objects it stores (thus the "T" at the beginning of the
+  name; "petra" being the word for "rock" or "foundation" since the rest of
+  Trilinos builds on these packages). deal.II is wrapping the Tpetra classes
+  in namespace LinearAlgebra::TpetraWrappers, but this process is still
+  ongoing and may be available in a version of deal.II after release 9.5.
```
Comment (kronbichler) [inline examples/step-40/doc/intro.dox:151]: What do you think about adding a short extension like this:
```suggestion
  ongoing and may be available in a version of deal.II after release 9.5.
- deal.II also implements a subset of parallel linear algebra functionality
  internally through the LinearAlgebra::distributed::Vector class (compatible
  with most Trilinos classes) and matrix-free linear solvers, which are run
  on some of the biggest machines in the world as well. For further details, see
  the step-37 tutorial program.
```

**Review by kronbichler — APPROVED** [review]
some small comments/suggestions

File: examples/step-40/doc/intro.dox (hunk 3 above)
Comment (bangerth) [reply examples/step-40/doc/intro.dox:151]: Ah yes, good point. I meant to mention this and forgot about it.

**Review by kronbichler — APPROVED** [review]

File: examples/step-40/doc/intro.dox [hunk 4]
```diff
@@ -93,6 +94,76 @@ It is probably worthwhile reading it for background information on how
...
+the current program is the first one that *really* covers parallel computing,
+it is probably worth giving a broad overview of parallel linear algebra here
+as well.
+
+First, the general mantra mentioned above was that *everything* has to be
+distributed. It does not scale if one process (or in fact *all* processes)
+have to keep a complete triangulation or even a substantial share of it; it
+all only works if every one of the $N$ processes in the parallel universe
+keep at most a small multiple of one $N$th of the triangulation.
+Similarly, each process can only hold a small multipe of one $N$th of
```
Comment (marcfehling) [inline examples/step-40/doc/intro.dox:109]:
```suggestion
Similarly, each process can only hold a small multiple of one $N$th of
```

**Review by marcfehling — APPROVED** [review]
Just one more typo

**Pull request merged**